plugins/yallah/src/main/java/de/dfki/vsm/xtension/yallah/YallahExecutor.java

```diff
@@ -0,0 +1,301 @@
+    private WsConnectContext mSocketConnectCtx = null;
+
+    /** The incremental counter used as unique ID for the sentences to speak, and as thread identifier. */
+    private int mSentenceCounter = 0 ;
+
+    /** The map of activity workers. This is used to keep track of threads waiting for the sentences to be spoken on the client.
+     * The key is a unique identifier of the spoken text.
+     * The value is the reference to the thread waiting inside the `execute()` method.
+     */
+    private final Map<String, ActivityWorker> mActivityWorkerMap = new HashMap<>();
+
+
+    public YallahExecutor(PluginConfig config, RunTimeProject project)
+    {
+        super(config, project);
+    }
+
+    private final static String MARKER = "$" ;
+
+    @Override
+    public String marker(long id)
+    {
+        return MARKER+id;
+    }
+
+
+    @Override
+    public void launch() {
+        mLogger.message("YALLAH plugin launching...");
+
+        //
+        // Setup the WebSocket server to get messages from a YALLAH application.
+        final int port = Integer.parseInt(Objects.requireNonNull(mConfig.getProperty("port")));
+        mLogger.message("Starting the WebSocket server on port " + port + "...");
+
+        //mWebSocketServer = Javalin.create(config -> config.enforceSsl = true).start(port);
+        mWebSocketServer = Javalin.create().start(port);
+
+        // This sequence sets up what the websocket server must do when events occur
+        mWebSocketServer.ws("/", ws -> {
+
+            // Client Connection
+            ws.onConnect(ctx -> {
+                mLogger.message("Getting connection request.");
+                mSocketConnectCtx = ctx;
+            });
+
+            // Client Message
+            ws.onMessage(this::handleMessage);
+
+            // Client disconnection
+            ws.onClose(ctx -> {
+                mSocketConnectCtx = null;
+
+                mLogger.message("Socket closed");
+
+                mLogger.message("Remove active (but not needed anymore) activity actions");
+                synchronized (mActivityWorkerMap) {
+                    mActivityWorkerMap.clear();
+                    // wake me up ..
+                    mActivityWorkerMap.notifyAll();
+                }
+            });
+
+            // Server error
+            ws.onError(ctx -> {
+                mLogger.failure("Error handling ws message exchange:" + ctx) ;
+                mLogger.message("Remove active activity actions to avoid deadlocks...");
+                synchronized (mActivityWorkerMap) {
+                    mActivityWorkerMap.clear();
+                    // wake me up ..
+                    mActivityWorkerMap.notifyAll();
+                }
+
+            });
+        });
+
+
+        //
+        //
+        String launch_mode = mConfig.getProperty("launchMode") ;
+        YallahProperties.LaunchMode lm = YallahProperties.LaunchMode.valueOf(launch_mode) ;
+        switch (lm) {
```

Comment:
Todos are still open. How does yallah launch at the moment?

Reply:
At the moment you have to launch the YALLAH application manually. It will automatically connect through a retry loop.